File: front/src/types/timeEntry.ts
export interface TimeEntry {
  id: number;
  taskId: number;
  taskTitle: string;
  projectName: string;
  userId: number;
  userName: string;
  date: string;
  hours: number;
  description?: string;
  createdAt?: string;
}

export interface ProjectHoursSummary {
  projectId: number;
  projectName: string;
  companyName: string;
  status: string;
  estimatedHours: number;
  loggedHours: number;
  deltaHours: number;
  burnPercent: number;
  totalEntries: number;
}

File: back/CRMF360.Domain/Entities/TimeEntry.cs
namespace CRMF360.Domain.Entities;

public class TimeEntry
{
    public int Id { get; set; }
    public int TaskId { get; set; }
    public int UserId { get; set; }
    public DateTime Date { get; set; }
    public decimal Hours { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Task Task { get; set; } = null!;
    public User User { get; set; } = null!;
}
